include/fastdds/dds/publisher/qos/DataWriterQos.hpp

```diff
@@ -58,6 +58,12 @@ class RTPSReliableWriterQos
                (this->disable_heartbeat_piggyback == b.disable_heartbeat_piggyback);
     }
 
+    inline void clear()
+    {
+        RTPSReliableWriterQos reset = RTPSReliableWriterQos();
```

Comment:
I don't love this. Maybe it would be better to implement a move assignment operator and to have the code for `clear` to be simply `*this = RtpsReliableWriterQos();`.